src/pages/index.js

```diff
@@ -88,81 +96,95 @@ const formValidationConfig = {
+  return api.updateUserAvatar(values).then((res) => {
+    const result = {
+      avatar: values.avatar,
+          };
+
+    userInfo.setAvatar(values.avatar);
+  });
+});
+// Opens the form for Avatar
+openAvatarButton.addEventListener("click", () => {
+  editAvatarValidator.toggleButtonState();
+  const avatarData = userInfo.getAvatar();
+  avatarInput.value = avatarData.src;
+
+  editAvatarForm.openModal();
+});
+
+// Opens the form for name and profession
 openEditButton.addEventListener("click", () => {
   editFormValidator.toggleButtonState();
   const profileData = userInfo.getUserInfo();
-
   nameInput.value = profileData.name;
   professionInput.value = profileData.profession;
+
   editProfileForm.openModal();
 });
 
-/* -------------------------------------------------------------------------- */
-/*                                  NEW CARD                                  */
-/* -------------------------------------------------------------------------- */
-
-const renderCard = (item) => {
-  const card = new Card({ item }, "#card-template", (title, link) => {
-    cardImageModal.openModal(title, link);
-  });
-  const cardElement = card.getCardElement();
-
-  return cardElement;
-};
-
+// For gaining access to methods within popup form
 const newDestinationCardForm = new PopupWithForm(
   "#newCard-modal",
   (newCardInputs) => {
-    const card = renderCard(newCardInputs);
-    defaultDestinationSection.addItem(card);
-
-    //renderCard(newCardInputs);
-    //const title = titleInput.value;
-    //const link = imageInput.value;
-    // const card = new Card({ title, link }, "#card-template");
-    // const cardElement = card.getCardElement();
-    // cardListEl.prepend(cardElement);
+    
+    return api.addDestinationCard(newCardInputs).then((res) => {
+     const card = renderCard(res);
+      defaultDestinationSection.addItem(card);
+      
+      
+    });
   }
 );
 
+// Opens the modal that adds destination cards
 openAddButton.addEventListener("click", () => {
   addFormValidator.toggleButtonState();
   newDestinationCardForm.openModal();
 });
 
-/* -------------------------------------------------------------------------- */
-/*                              CARD IMAGE POP UP                             */
-/* -------------------------------------------------------------------------- */
+// Creates an instance of a delete confirmation
+const deleteCardConfirm = new PopupWithConfirm(
+  //modalSelector
+  "#deleteCard-modal",
+  // handleConfirmSubmit
+
+  // deleteCard()
+  (card, cardId) => {
+    return api.delDestinationCard(cardId).then(() => {
+      card.handleRemoveCard();
+    });
+  }
+);
 
 const cardImageModal = new PopupWithImage("#viewImage-modal");
 
 const addFormValidator = new FormValidator(formValidationConfig, addCardForm);
 const editFormValidator = new FormValidator(formValidationConfig, editCardForm);
+const editAvatarValidator = new FormValidator(formValidationConfig, avatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```

Reply:
Leaving comment open and taking improvement under advisement for future updates